Frameworks/UIKit/UIApplicationMain.mm

```diff
@@ -140,6 +140,25 @@ int UIApplicationMainInit(NSString* principalClassName,
     [[NSThread currentThread] _associateWithMainThread];
     NSDictionary* infoDict = [[NSBundle mainBundle] infoDictionary];
 
+    // Register fonts listed in app's Info.plist, from the app's bundle
+    // This needs to happen before [UIApplication new]/[objc_getClass(pClassName) new] below,
+    // as they may attempt to use fonts from the app's bundle
+    if (infoDict) {
+        NSArray* fonts = [infoDict objectForKey:@"UIAppFonts"];
+        if (fonts != nil) {
+            NSMutableArray* fontURLs = [NSMutableArray array];
+            for (NSString* curFontName in fonts) {
+                // curFontName contains extension, so pass in nil
+                NSURL* url = [[NSBundle mainBundle] URLForResource:curFontName withExtension:nil];
+                if (url != nil) {
+                    [fontURLs addObject:url];
+                }
+            }
+
+            CTFontManagerRegisterFontsForURLs((CFArrayRef)fontURLs, kCTFontManagerScopeNone, nil);
+        }
+    }
+
```

Comment:
the auto release pools that are created after this must instead be created first.  In fact, the pools must be created before the [NSThread] line.